src/Workspaces/Core/Portable/Workspace/Solution/LoadableTextAndVersionSource.cs

```diff
@@ -16,12 +17,26 @@ internal sealed class LoadableTextAndVersionSource : ITextAndVersionSource
     private sealed class LazyValueWithOptions
     {
         public readonly LoadableTextAndVersionSource Source;
-        public readonly AsyncLazy<TextAndVersion> LazyValue;
```

Comment:
this was the only place where we used an AsyncLazy with `cacheResult: false`.  Just inlined the logic here to simplify everywhere else.

Reply:
Note: this type is pretty suspect to begin with.  As it has mutable semantics.  Namely, it expected to be able to dump the value from teh async-lazy from before and load a *different* value (With non-equal semantics).  That definitely isn't in the spirit of the lazy computation, as normally the idea would be that if it recomptued you'd get the same effective result as before.

So, to me, this is just preferable for this type, even if AsyncLazy still was able to have the knob to not cache values.